src/main/java/software/amazon/encryption/s3/S3AsyncEncryptionClient.java

```diff
@@ -489,6 +511,121 @@ public Builder secureRandom(SecureRandom secureRandom) {
             return this;
         }
 
+        /**
+         * The credentials provider to use for all inner clients, including KMS, if a KMS key ID is provided.
+         * @param awsCredentialsProvider
+         * @return
+         */
+        public Builder credentialsProvider(AwsCredentialsProvider awsCredentialsProvider) {
+            _awsCredentialsProvider = awsCredentialsProvider;
+            return this;
+        }
+
+        /**
+         * The AWS region to use for all inner clients, including KMS, if a KMS key ID is provided.
+         * @param region
+         * @return
+         */
+        public Builder region(Region region) {
+            _region = region;
+            return this;
+        }
+
+        public Builder dualstackEnabled(Boolean isDualStackEnabled) {
```

Comment:
What's going on with this and fipsEnabled? Do we use these? Or are we just acting as a passthrough to send these to SDK? How does SDK use these?

I might want some more Javadocs here just to explain how this is used, esp. on the client builder

Reply:
It's just a pass-through, 
my understanding is that AWS services have both a "default" endpoint,
and a FIPS endpoint, 
which only supports TLS ciphers that are FIPS-validated. 
So this points the SDK to the FIPS one. 
In that sense, it's similar to endpoint override, but for FIPS. 

We should clarify that it does not mean that the S3EC will use FIPS for its encryption/decryption, I'll add a comment.